File: hello.py
#!/usr/bin/env python3
"""
A simple Python script demonstrating basic functionality.
"""

def greet(name="World"):
    """Return a greeting message."""
    return f"Hello, {name}!"

def calculate_fibonacci(n):
    """Calculate the nth Fibonacci number."""
    if n <= 1:
        return n
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

def main():
    """Main function demonstrating the script features."""
    print(greet())
    print(greet("Python"))
    
    print("\nFibonacci sequence (first 10 numbers):")
    for i in range(10):
        print(f"F({i}) = {calculate_fibonacci(i)}")

if __name__ == "__main__":
    main()

File: hello.js
/**
 * A simple JavaScript script demonstrating basic functionality.
 */

function greet(name = "World") {
    return `Hello, ${name}!`;
}

function calculateFactorial(n) {
    if (n <= 1) return 1;
    return n * calculateFactorial(n - 1);
}

function isPrime(num) {
    if (num <= 1) return false;
    if (num <= 3) return true;
    if (num % 2 === 0 || num % 3 === 0) return false;
    
    for (let i = 5; i * i <= num; i += 6) {
        if (num % i === 0 || num % (i + 2) === 0) return false;
    }
    return true;
}

function main() {
    console.log(greet());
    console.log(greet("JavaScript"));
    
    console.log("\nFactorials (1-10):");
    for (let i = 1; i <= 10; i++) {
        console.log(`${i}! = ${calculateFactorial(i)}`);
    }
    
    console.log("\nPrime numbers up to 50:");
    for (let i = 2; i <= 50; i++) {
        if (isPrime(i)) {
            process.stdout.write(`${i} `);
        }
    }
    console.log();
}

// Run if this script is executed directly
if (require.main === module) {
    main();
}

module.exports = { greet, calculateFactorial, isPrime };